certora/specs/ERC20FlashMint.spec

```diff
@@ -1,29 +1,29 @@
-import "helpers/helpers.spec"
-import "methods/IERC20.spec"
-import "methods/IERC3156.spec"
+import "helpers/helpers.spec";
+import "methods/IERC20.spec";
+import "methods/IERC3156.spec";
 
 methods {
     // non standard ERC3156 functions
-    flashFeeReceiver() returns (address) envfree
+    function flashFeeReceiver() external returns (address) envfree;
 
     // function summaries below
-    _mint(address account, uint256 amount)              => specMint(account, amount)
-    _burn(address account, uint256 amount)              => specBurn(account, amount)
-    _transfer(address from, address to, uint256 amount) => specTransfer(from, to, amount)
+    function _._mint(address account, uint256 amount)              internal => specMint(account, amount)        expect void ALL;
+    function _._burn(address account, uint256 amount)              internal => specBurn(account, amount)        expect void ALL;
+    function _._transfer(address from, address to, uint256 amount) internal => specTransfer(from, to, amount)   expect void ALL;
```

Comment:
How is that affected by `_update` ?

Reply:
It looks like we should be "overriding" `_update` instead of these 3 functions.

One thing I don't understand is, is this completely replacing the logic of these three functions? It sounds like we would want to add ghosts on top of the default functionalit but if I understand correctly we're replacing the internal logic completely.

Reply:
> is this completely replacing the logic of these three functions?

Yes.

I replaced these 3 summaries with only one `_._update` summary and it worked. Also tried removing it but the counter-example is similar to what we've seen before where the contract starts with arbitrary storage values. I have a hunch that we can remove this summarization by using a few `require` statements, but as far as I'm concerned, the summarization is fine because `_update` is already proven in the ERC20 spec.